### PR TITLE
Auto install dependencies

### PR DESCRIPTION
**Pull request by sindresorhus**
Many users have voiced concern of there being too many commands to get started. The original intention of not automatically installing deps after scaffolding was for clearity of what was happening. Since users would think a npm bug was a bug with Yeoman. Seeing as this has helped nothing, and people keep opening invalid issues like crazy, I don't think that is an argument anymore.

After thinking about this for a while, I've come to the conclusion that we can do both clearity and ease of use, by doing it automatically, but being totally clear about it. That means showing a clear message as shown below and documenting it. After all, Yeoman is about making front-end development less painful.

@addyosmani @sleeper @btford @passy wdyt?

_this also parallizes both commands, so it's faster than before._

Screen Shot 2013-03-17 at 16 04 58

### CODE REVIEW

**Comment by passy** [comment]
I've been thinking about this myself and had the same concerns that users could presume errors in the process would be caused by yeoman. I think this is a very elegant solution to that problem.

File: app/index.js [hunk 1]
```diff
@@ -25,7 +26,9 @@ function AppGenerator(args, options, config) {
   this.mainCoffeeFile = 'console.log "\'Allo from CoffeeScript!"';
 
   this.on('end', function () {
-    console.log('\nI\'m all done. Just run ' + 'npm install && bower install'.bold.yellow + ' to install the required dependencies.');
+    console.log('\n\nI\'m all done. Running ' + 'npm install &; bower install'.bold.yellow + ' for you to install the required dependencies. If this fails, try running the command yourself.\n\n');
```
Comment (passy) [inline app/index.js]: Should be just `npm install & bower install`. The semicolon there is a syntax error, if I'm not mistaken.

Comment (sindresorhus) [inline app/index.js]: You were the one that suggested that syntax.

Comment (passy) [inline app/index.js]: I know. :persevere: Searching for the other occurrences right now.

Comment (passy) [inline app/index.js]: It works in fish, but not in bash.

Comment (sindresorhus) [inline app/index.js]: whatever, `npm install & bower install` is one char shorter, so why not. Also Bash sucks.

**Comment by sindresorhus** [comment]
the ping machine pings @addyosmani @sleeper @btford ping pong pang

## 

// @paulirish @kevva

**Comment by btford** [comment]
I like it. Make it so!

**Comment by kevva** [comment]
Pong, looks good! One could argue that the error output by npm and bower should be enough to make a distinction between those and yeoman. Also, parallelizing is a big plus.

**Comment by sindresorhus** [comment]
@addyosmani @paulirish last chance to object ;)

**Comment by btford** [comment]
No objections. Do it.

**Comment by addyosmani** [comment]
I'm 100% behind this change. We should definitely do it.

**Comment by addyosmani** [comment]
PS: We need to make sure that if/when this lands we update all of the installation guides for relevant generators as the CLI will be providing instructions for the two steps we're aiming to automate here.

**Comment by passy** [comment]
Yeah! :space_invader:

**Comment by sindresorhus** [comment]
@addyosmani ticket: https://github.com/yeoman/yeoman/issues/1007

**Comment by yveslaroche** [comment]
I get the following error after running "yo webapp" now on Windows using the Git Bash. 

```
events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: spawn ENOENT
    at errnoException (child_process.js:977:11)
    at Process.ChildProcess._handle.onexit (child_process.js:768:34)
```

However, running "npm install & bower install" manually works just fine.

**Comment by sindresorhus** [comment]
@yveslaroche can you gist your full terminal output? what Node version?

@passy @kevva can you test?

**Comment by yveslaroche** [comment]
I'm using node.js v0.10.3. My full terminal output is here: https://gist.github.com/yveslaroche/5337427

I've also linked in the "generator" repo with the "generator-webapp", as it is a dependency there is well. This step is missing from the contributing instructions. I noticed this because I had the following error before, which was fixed with the latest commit to the generator.

```
D:\Dropbox\Dev\GitHub\yeoman\generator-webapp\node_modules\yeoman-generator\node
_modules\bower\node_modules\tmp\lib\tmp.js:219
    throw err;
          ^
Error: spawn ENOENT
    at errnoException (child_process.js:977:11)
    at Process.ChildProcess._handle.onexit (child_process.js:768:34)
```

**Comment by kevva** [comment]
@yveslaroche, try clearing your bower cache and make sure you've got the latest version of bower. May be related to twitter/bower#307.

**Comment by yveslaroche** [comment]
@kevva I've cleared bower cache before. And just tried it again with bower v0.8.6 and the latest bower master branch, too. I'm still getting the same error https://gist.github.com/yveslaroche/5337427 with both bower versions.

**Comment by addyosmani** [comment]
hmm. Haven't been able to reproduce. Anyone else have better luck?

**Pull request merged**